xdsl/dialects/arith.py

```diff
@@ -127,6 +136,55 @@ def get(operand1: Union[Operation, SSAValue],
         return Subi.build(operands=[operand1, operand2],
                           result_types=[operand1.typ])
 
+@irdl_op_definition
+class DivUI(Operation):
+    """
+    Unsigned integer division. Rounds towards zero. Treats the leading bit as
+    the most significant, i.e. for `i16` given two's complement representation,
+    `6 / -2 = 6 / (2^16 - 2) = 0`.
+    """
+    name: str = "arith.divui"
+    lhs = OperandDef(signlessIntegerLike)
+    rhs = OperandDef(signlessIntegerLike)
+    result = ResultDef(signlessIntegerLike)
+
+    # TODO replace with trait
+    def verify_(self) -> None:
+        if self.lhs.typ != self.rhs.typ or self.rhs.typ != self.result.typ:
+            raise VerifyException(
+                "expect all input and result types to be equal")
+
+    @staticmethod
+    def get(operand1: Union[Operation, SSAValue],
+            operand2: Union[Operation, SSAValue]) -> DivUI:
+        operand1 = SSAValue.get(operand1)
+        return DivUI.build(operands=[operand1, operand2],
+                           result_types=[operand1.typ])
+
+
+@irdl_op_definition
+class DivSI(Operation):
+    """
+    Signed integer division. Rounds towards zero. Treats the leading bit as
+    sign, i.e. `6 / -2 = -3`.
+    """
+    name: str = "arith.divsi"
+    lhs = OperandDef(signlessIntegerLike)
+    rhs = OperandDef(signlessIntegerLike)
+    result = ResultDef(signlessIntegerLike)
+
+    # TODO replace with trait
+    def verify_(self) -> None:
+        if self.lhs.typ != self.rhs.typ or self.rhs.typ != self.result.typ:
```

Comment:
I did this check similar to how it was done for the other ops in this file

`any` takes one array of boolean values as argument. To make it work here something like this 
```Python
if any(elem != self.rhs.typ for elem in [self.result.typ, self.lhs.typ])
```
 is needed. 
I think this is less readable in the case of only two comparisons. 
I am still happy to change the verifiers in this file to this or something better if we come up with something.

Reply:
Yes, I also don't think it's not much worth thinking about this too much, since I'll soon introduce a patch that will remove this part of the code entirely!